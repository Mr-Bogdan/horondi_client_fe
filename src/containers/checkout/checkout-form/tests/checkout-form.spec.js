import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import CheckoutForm from '../checkout-form';
import Delivery from '../delivery/delivery';
import DeliveryType from '../../delivery-type/delivery-type';

jest.mock('../checkout-form.styles', () => ({ useStyles: () => ({ Theme: 'lightMode' }) }));
jest.mock('../../delivery-type/delivery-type.styles', () => ({ useStyles: () => ({}) }));
jest.mock('react-redux');
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: () => ({ paymentMethod: 'card' }),
    i18n: { language: 'en' }
  })
}));

const myOnSubmit = jest.fn().mockResolvedValueOnce({ data: { paymentMethod: 'card' } });
const dispatch = jest.fn();
const props = {
  language: 1,
  isLightTheme: true,
  currency: 0,
  cartItems: [{ price: [{ currency: 'ua', value: 100 }] }],
  deliveryType: 'NOVAPOSTCOURIER'
};
const userData = {
  loginError: '',
  userLoading: false,
  language: 0,
  snackBarStatus: '',
  snackBarSeverity: '',
  snackBarMessage: ''
};

useDispatch.mockImplementation(() => dispatch);
useSelector.mockImplementation(() => userData);

describe('CheckoutForm component tests', () => {
  it('should contain <CheckoutForm /> component <Delivery />', () => {
    const wrapper = shallow(<CheckoutForm {...props} />);
    expect(wrapper.find(Delivery).length).toEqual(1);
    expect(wrapper.find(Link).at(1).props().children[1].paymentMethod).toEqual('card');
  });
  it('should match snapshot', () => {
    const wrapper = shallow(<CheckoutForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should submit add payment method', async () => {
    const wrapper = shallow(<CheckoutForm {...props} onSubmit={myOnSubmit} />);
    wrapper.find('form').simulate('submit');
  });
  it('should contain <CheckoutForm /> component <DeliveryType />', () => {
    const wrapper = shallow(<CheckoutForm {...props} language={0} />);
    expect(wrapper.find(DeliveryType).length).toEqual(1);
  });
});
