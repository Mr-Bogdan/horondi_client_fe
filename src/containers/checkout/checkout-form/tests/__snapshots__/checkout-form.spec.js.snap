// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutForm component tests should match snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <div>
        <div>
          <Link
            to="/cart"
          >
            <Memo(ForwardRef(KeyboardBackspaceIcon))
              color="action"
            />
          </Link>
        </div>
        <h2>
          <Component />
        </h2>
        <div />
      </div>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <div>
          <h3>
            <Component />
          </h3>
          <div>
            <div
              key="firstName"
            >
              <WithStyles(ForwardRef(TextField))
                data-cy="firstName"
                label="First name *"
                name="firstName"
                onChange={[Function]}
                size="small"
                value=""
                variant="outlined"
              />
            </div>
            <div
              key="lastName"
            >
              <WithStyles(ForwardRef(TextField))
                data-cy="lastName"
                label="Last name *"
                name="lastName"
                onChange={[Function]}
                size="small"
                value=""
                variant="outlined"
              />
            </div>
          </div>
          <div>
            <div
              key="email"
            >
              <WithStyles(ForwardRef(TextField))
                data-cy="email"
                label="Email *"
                name="email"
                onChange={[Function]}
                size="small"
                value=""
                variant="outlined"
              />
            </div>
            <div
              key="phoneNumber"
            >
              <WithStyles(ForwardRef(TextField))
                data-cy="phoneNumber"
                label="Phone *"
                name="phoneNumber"
                onChange={[Function]}
                size="small"
                value=""
                variant="outlined"
              />
            </div>
          </div>
        </div>
        <DeliveryType />
        <Delivery
          deliveryType="NOVAPOSTCOURIER"
          errors={Object {}}
          handleChange={[Function]}
          isLightTheme={true}
          language={1}
          setFieldValue={[Function]}
          touched={Object {}}
          values={
            Object {
              "city": "",
              "cityId": "",
              "courierOffice": "",
              "district": "",
              "districtId": "",
              "email": "",
              "firstName": "",
              "flat": "",
              "house": "",
              "lastName": "",
              "paymentMethod": "",
              "phoneNumber": "",
              "region": "",
              "regionId": "",
              "street": "",
              "userComment": "",
            }
          }
        />
        <div>
          <h2>
            <Component />
          </h2>
          <WithStyles(ForwardRef(FormControl))
            variant="outlined"
          >
            <WithStyles(ForwardRef(InputLabel))
              variant="outlined"
            >
              <Component />
            </WithStyles(ForwardRef(InputLabel))>
            <WithStyles(ForwardRef(Select))
              data-cy="paymentMethod"
              label={
                Object {
                  "paymentMethod": "card",
                }
              }
              name="paymentMethod"
              onChange={[Function]}
              value=""
            >
              <WithStyles(ForwardRef(MenuItem))
                key="card"
                value="card"
              >
                card
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </div>
        <div>
          <h2>
            <Component />
          </h2>
          <div>
            <WithStyles(ForwardRef(TextField))
              data-cy="userComment"
              multiline={true}
              name="userComment"
              onChange={[Function]}
              rows={4}
              size="small"
              value=""
              variant="outlined"
            />
          </div>
          <p>
            <Component />
          </p>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <div>
          <h2>
            <Component />
          </h2>
          <div />
        </div>
        <div>
          <div>
            <h4>
              <Component />
            </h4>
            <p
              className="undefined undefined"
            >
              100
              Â 
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      384,
                      512,
                      Array [],
                      "f6f2",
                      "M368 240c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16h-41.86c13.41-28.63 13.74-63.33-4.13-94.05C303.34 49.84 267.1 32 229.96 32h-78.82c-24.32 0-47.86 8.53-66.54 24.09L72.83 65.9c-10.18 8.49-11.56 23.62-3.07 33.8l20.49 24.59c8.49 10.19 23.62 11.56 33.81 3.07l11.73-9.78c4.32-3.6 9.77-5.57 15.39-5.57h83.62c11.69 0 21.2 9.52 21.2 21.2 0 5.91-2.48 11.58-6.81 15.58L219.7 176H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h134.37l-34.67 32H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h41.86c-13.41 28.63-13.74 63.33 4.13 94.05C80.66 462.15 116.9 480 154.04 480h78.82c24.32 0 47.86-8.53 66.54-24.09l11.77-9.81c10.18-8.49 11.56-23.62 3.07-33.8l-20.49-24.59c-8.49-10.19-23.62-11.56-33.81-3.07l-11.75 9.8a23.992 23.992 0 0 1-15.36 5.56H149.2c-11.69 0-21.2-9.52-21.2-21.2 0-5.91 2.48-11.58 6.81-15.58L164.3 336H368c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16H233.63l34.67-32H368z",
                    ],
                    "iconName": "hryvnia",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              />
            </p>
          </div>
          <button
            type="submit"
          >
            Confirm order
          </button>
          <div>
             
            <Component />
            <Link
              rel="noreferrer"
              target="_blank"
              to="/pages/user-agreement"
            >
               
              <Component />
               
            </Link>
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </form>
</div>
`;
